pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -13,146 +13,411 @@
+This module contains the :class:`~.LightningKokkos` class, a PennyLane simulator device that
 interfaces with C++ for fast linear algebra calculations.
 """
-
-from os import getenv
+from dataclasses import replace
+from numbers import Number
 from pathlib import Path
-from typing import List
-from warnings import warn
+from typing import Callable, Optional, Sequence, Tuple, Union
 
 import numpy as np
 import pennylane as qml
-from pennylane import BasisState, DeviceError, QuantumFunctionError, Rot, StatePrep, math
-from pennylane.measurements import Expectation, MidMeasureMP, State
-from pennylane.ops import Conditional
-from pennylane.ops.op_math import Adjoint
-from pennylane.wires import Wires
-
-from pennylane_lightning.core._serialize import QuantumScriptSerializer, global_phase_diagonal
-from pennylane_lightning.core._version import __version__
-from pennylane_lightning.core.lightning_base import LightningBase, _chunk_iterable
+from pennylane.devices import DefaultExecutionConfig, Device, ExecutionConfig
+from pennylane.devices.default_qubit import adjoint_ops
+from pennylane.devices.modifiers import simulator_tracking, single_tape_support
+from pennylane.devices.preprocess import (
+    decompose,
+    mid_circuit_measurements,
+    no_sampling,
+    validate_adjoint_trainable_params,
+    validate_device_wires,
+    validate_measurements,
+    validate_observables,
+)
+from pennylane.measurements import MidMeasureMP
+from pennylane.operation import DecompositionUndefinedError, Operator, Tensor
+from pennylane.ops import Prod, SProd, Sum
+from pennylane.tape import QuantumScript, QuantumTape
+from pennylane.transforms.core import TransformProgram
+from pennylane.typing import Result, ResultBatch
+
+from ._state_vector import LightningStateVector
 
 try:
     # pylint: disable=import-error, no-name-in-module
-    from pennylane_lightning.lightning_kokkos_ops import (
-        InitializationSettings,
-        MeasurementsC64,
-        MeasurementsC128,
-        StateVectorC64,
-        StateVectorC128,
-        allocate_aligned_array,
-        backend_info,
-        print_configuration,
-    )
-
-    # pylint: disable=import-error, no-name-in-module, ungrouped-imports
-    from pennylane_lightning.lightning_kokkos_ops.algorithms import (
-        AdjointJacobianC64,
-        AdjointJacobianC128,
-        create_ops_listC64,
-        create_ops_listC128,
-    )
+    from pennylane_lightning.lightning_kokkos_ops import backend_info, print_configuration
 
     LK_CPP_BINARY_AVAILABLE = True
 except ImportError:
     LK_CPP_BINARY_AVAILABLE = False
     backend_info = None
 
+Result_or_ResultBatch = Union[Result, ResultBatch]
+QuantumTapeBatch = Sequence[QuantumTape]
+QuantumTape_or_Batch = Union[QuantumTape, QuantumTapeBatch]
+PostprocessingFn = Callable[[ResultBatch], Result_or_ResultBatch]
+
+
+def simulate(  # pylint: disable=unused-argument
+    circuit: QuantumScript,
+    state: LightningStateVector,
```

Comment:
```suggestion
    state: LightningKokkosStateVector,
```

Reply:
Since this is just a reference to `._state_vector`, don't we want to keep it generic for now? I'm thinking down the line we may want to have a base class with an agnostic reference to the state vector class. Not sure how much we would gain in `simulate`, but I think there is quite a bit a code duplication to be avoided this way. What do you think?

Reply:
The plan for the future is to merge all similar classes but for sake of time we will duplicate them for now